relay_rpc/src/domain.rs

```diff
@@ -119,24 +119,24 @@ impl DecodedClientId {
     }
 
     #[inline]
-    pub fn from_key(key: &PublicKey) -> Self {
+    pub fn from_key(key: &VerifyingKey) -> Self {
         Self(*key.as_bytes())
     }
 
     #[inline]
-    pub fn as_public_key(&self) -> PublicKey {
+    pub fn as_public_key(&self) -> VerifyingKey {
         // We know that the length is correct, so we can just unwrap.
-        PublicKey::from_bytes(&self.0).unwrap()
+        VerifyingKey::from_bytes(&self.0).unwrap()
```

Comment:
This function only accepts correct-length arrays, so this comment is wrong. This function may panic for other reasons, we should probably investigate is using `unwrap()` is dangerous here

Reply:
You're right. It can panic if the bytes are invalid. I think the easiest and non-breaking solution here is to add `try_as_public_key()` method which wouldn't unwrap. And also adding a doc for `as_public_key()` that it will panic on invalid key.

Reply:
Leaving it as-is since this bug already existed, but made [an issue](https://github.com/WalletConnect/WalletConnectRust/issues/66) to track it